go/encoding/basex/encoding.go

```diff
@@ -41,15 +42,22 @@ func NewEncoding(encoder string, inBlockLen int) *Encoding {
 	// output block length.  We need to round up to fit the overflow.
 	outBlockLen := int(math.Ceil(float64(8*inBlockLen) / logOfBase))
 
+	// maxEncodedBitsPerBlock determines the maximum number of bits
+	// we can fit into an encoded block of the given block length.
+	// We're actually not going to be using this many bits, but we
+	// need to know for the purposes of left shifting during encoding.
+	maxEncodedBitsPerBlock := int(math.Floor(logOfBase * float64(outBlockLen)))
```

Comment:
Ah never mind. I was missing that the number of extra bits is different for short blocks.